client/state/sites/selectors.js

```diff
@@ -550,3 +553,307 @@ export function getSitePostsPage( state, siteId ) {
 export function getSiteFrontPageType( state, siteId ) {
 	return getSiteOption( state, siteId, 'show_on_front' );
 }
+
+/**
+ * Determines if a Jetpack site has opted in for full-site management
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {?Boolean} if the site can be managed from calypso
+ */
+export function canJetpackSiteManage( state, siteId ) {
+	// for versions 3.4 and higher, canManage can be determined by the state of the Manage Module
+	const siteJetpackVersion = getSiteOption( state, siteId, 'jetpack_version' );
+
+	if ( ! siteJetpackVersion ) {
+		return null;
```

Comment:
I'm not sure, I tried to stay as close from the original implementation as possible.
Maybe at some point a jetpack site can be connected but all the information has not been synchronized yet. I haven't had a look into the jetpack components yet but they might just use this behaviour.